docs/object-utils.md

```diff
@@ -24,6 +24,10 @@ Returns an object which contains fields specified in `propNames` with the same v
 
 Returns true if `params` has no own properties or only own properties with value `undefined`, false otherwise.
 
-`groupBy<T>(inputArray: T[], propName: string): Record<string, T[]>`
+`export function groupBy<T extends { [K in keyof T]: string | number | symbol | null | undefined }, K extends keyof T>(array: T[], selector: K): Record<string | number | symbol, T[]>`
 
-Returns an object composed of keys generated from the values of `propName` field for the `inputArray` elements. The order of grouped values is determined by the order they occur in collection.
+The `groupBy` function takes an array of objects and a `selector`, groups the objects based on selected key and returns an object with unique keys from the selector and corresponding groups as arrays.
+
+`export function groupByUnique<T extends { [K in keyof T]: string | number | symbol | null | undefined }, K extends keyof T>(array: T[], selector: K): Record<string | number | symbol, T>`
+
+Similar to `groupBy`, but the value is a single element which corresponds to the last found element in the array
```

Comment:
In order to avoid unpredictable state, I'd throw on non-unique values instead. if multiple values are ok, groupBy sounds like an option with more predictable behaviour

Reply:
Mmm, sounds good to me, I will address the change, thanks Igor :)